### PR TITLE
🚧  Add github actions automation to build/test/push the image

### PR DESCRIPTION
**Pull request by bakayolo**
- Upon PR opened to `main` branch, we should build and test the code / image.
- Upon merge into `main` branch, we should push the image into docker registry with `latest` tag.

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  main:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Build and push
```
Comment (nandiheath) [inline .github/workflows/docker.yml]: action - should rename to `Build`

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+jobs:
+  main:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
```
Comment (nandiheath) [inline .github/workflows/docker.yml:15]: question - is there a reason why we build cross-architecture?

**Review by nandiheath — CHANGES_REQUESTED** [review]
we should also add tag -> we tag on Github and the CI should tag the image built and push it to the registry
so on both Github and Dockerhub we will have the same artifact -> code and image

File: .github/workflows/docker.yml (hunk 2 above)
Comment (bakayolo) [reply .github/workflows/docker.yml:15]: If you mean https://github.com/docker/build-push-action#multi-platform-image, we don't

Comment (nandiheath) [reply .github/workflows/docker.yml:15]: then why we need QEMU and buildx, aren't they for cross platform docker build?

Comment (nandiheath) [reply .github/workflows/docker.yml:15]: suggest we either remove QEMU + buildx or we enable multiplatform builds -> at least amd64 + arm64 (support new mac M1 chips)

Comment (nandiheath) [reply .github/workflows/docker.yml:15]: after discussion agree to keep it like this first but once some user complaints we should add platforms

**Review by nandiheath — APPROVED** [review]

**Pull request merged**